api/service/syncing/syncing.go

```diff
@@ -540,7 +540,7 @@ func (ss *StateSync) updateBlockAndStatus(block *types.Block, bc *core.BlockChai
 	// Verify block signatures
 	// TODO chao: only when block is verified against last commit sigs, we can update the block and status
 	if block.NumberU64() > 1 {
-		err := core.VerifyBlockLastCommitSigs(bc, block.Header())
```

Comment:
Previously, we use most common block hashes to determine which block to download, then we use that blocks to try to update the status. Now from my understanding, here we are improving the logic by verify the block first before put it into the queue.  If that is the case, VerifyHeader call needs cooperate with the retry download other blocks logic before updateBlockAndStatus.